core/src/banking_stage.rs

```diff
@@ -1190,6 +1216,71 @@ impl BankingStage {
             .collect()
     }
 
+    fn generate_vote_packet_indexes(vers: &PinnedVec<Packet>) -> Vec<usize> {
+        vers.iter()
+            .enumerate()
+            .filter_map(|(index, ver)| {
+                if !ver.meta.discard {
+                    let tx: Transaction = limited_deserialize(&ver.data[0..ver.meta.size]).ok()?;
+                    if is_simple_vote_transaction(&tx) {
+                        Some(index)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn process_packets_vote(
+        mms: &mut Vec<Packets>,
+        vote_tx_count: &mut usize,
+        skip_process_packets_vote_count: &mut usize,
+        unprocessed_vote_tx_count: &mut usize,
+        poh: &Arc<Mutex<PohRecorder>>,
+        recorder: &TransactionRecorder,
+        transaction_status_sender: Option<TransactionStatusSender>,
+        gossip_vote_sender: &ReplayVoteSender,
+        banking_stage_stats: &BankingStageStats,
+    ) {
+        let mms_iter = mms.iter_mut();
+        for msgs in mms_iter {
+            let bank_start = poh.lock().unwrap().bank_start();
+            if PohRecorder::get_bank_still_processing_txs(&bank_start).is_none() {
+                // do noting here, the main loop will take care of unprocessed packets
+                *skip_process_packets_vote_count += 1;
+                continue;
+            }
+            let (bank, bank_creation_time) = bank_start.unwrap();
+
+            let packet_indexes = Self::generate_vote_packet_indexes(&msgs.packets);
+            let (processed, _verified_txs_len, unprocessed_indexes) =
```

Comment:
So if this fails due to something like lock contention, does the vote just enter the regular processing pipeline again?

Reply:
this is the case for now, I was bit concerned about its likelihood, added counter `banking_stage-loop-stats::unprocessed_vote_tx_count`, monitored it over few bench-tps tests, it is always zero. Feels like this is an unlikely event. 

I also don't have a clear path for handling unprocessed votes, maybe can put them on top of buffered packets, feels like a lot of changes, not super comfortable about it. 

![image](https://user-images.githubusercontent.com/82401714/133635591-1c4f2922-69d7-4f02-8769-982986689cf3.png)